### PR TITLE
Add notification channel claim value to /init-authenticator endpoint response

### PR DESCRIPTION
**Pull request by amanda-ariyaratne**
## Purpose
Previously in `/init-authenticator` API call did not return either mobile number or user email address along with response. This PR is having the ability to return the required user claim in API response as follows.

```
{
    "flowId": "867e3032-25d2-4f1f-bb84-611013d10c53",
    "authenticator": "SMSOTP",
    "notificationTarget": "******6511"
}
```

### Related Issues
- https://github.com/wso2/product-is/issues/19952

### Related PRs
- https://github.com/wso2-extensions/identity-oauth2-grant-rest/pull/12

### CODE REVIEW

File: components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java [hunk 1]
```diff
@@ -1073,4 +1087,95 @@ private String getFederatedIdpChannelName(String authenticatorName, RestAuthenti
...
+        AbstractUserStoreManager userStoreManager;
+        Map<String, String> userAttributes;
+        try {
+            userStoreManager = (AbstractUserStoreManager) AuthenticationServiceDataHolder.getInstance()
+                    .getRealmService().getTenantUserRealm(userTenantId).getUserStoreManager();
+            userAttributes = userStoreManager.getUserWithID(userId, new String[]{requestedClaimURI},
+                    UserCoreConstants.DEFAULT_PROFILE).getAttributes();
+
+            if(LOG.isDebugEnabled()) {
+                LOG.debug("Verifier claim value resolved for the user id : " + userId);
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java:1111]: let's add the resolved claim value as well for the verification channel in the log

File: components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java [hunk 2]
```diff
@@ -1073,4 +1087,95 @@ private String getFederatedIdpChannelName(String authenticatorName, RestAuthenti
...
+        try {
+            federatedAuthenticatorName = getFederatedIdpChannelName(authenticatorName, authContext);
+            if (Constants.AUTHENTICATOR_NAME_EMAILOTP.equals(federatedAuthenticatorName)) {
+                return StringUtils.isNotEmpty(AuthenticationServiceDataHolder.getConfigs().getEmailAddressRegex());
+            } else if (Constants.AUTHENTICATOR_NAME_SMSOTP.equals(federatedAuthenticatorName)) {
+                return StringUtils.isNotEmpty(AuthenticationServiceDataHolder.getConfigs().getMobileNumberRegex());
+            }
+            return false;
+        } catch (AuthenticationException e) {
+            return false;
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java:1150]: add a debug log here

File: components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java [hunk 3]
```diff
@@ -1073,4 +1087,95 @@ private String getFederatedIdpChannelName(String authenticatorName, RestAuthenti
...
+                emailAddressRegex = StringEscapeUtils.unescapeHtml(emailAddressRegex);
+                return notificationTarget.replaceAll(emailAddressRegex, Constants.MASKING_CHARACTER);
+            } else if (Constants.AUTHENTICATOR_NAME_SMSOTP.equals(federatedAuthenticatorName)) {
+                String mobileNumberRegex = AuthenticationServiceDataHolder.getConfigs().getMobileNumberRegex();
+                mobileNumberRegex = StringEscapeUtils.unescapeHtml(mobileNumberRegex);
+                return notificationTarget.replaceAll(mobileNumberRegex, Constants.MASKING_CHARACTER);
+            }
+            return StringUtils.EMPTY;
+        } catch (AuthenticationException e) {
+            return StringUtils.EMPTY;
```
Comment (pamodaaw) [inline components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java:1178]: add a debug log here

File: components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java (hunk 1 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java]: Fixed with 28d24d24dd8ae8ac604217dff5d2b45cd3140957

File: components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java (hunk 2 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java:1153]: Fixed with 28d24d24dd8ae8ac604217dff5d2b45cd3140957

File: components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java (hunk 3 above)
Comment (amanda-ariyaratne) [reply components/org.wso2.carbon.identity.oauth2.grant.rest.core/src/main/java/org/wso2/carbon/identity/oauth2/grant/rest/core/RestAuthenticationServiceImpl.java:1184]: Fixed with 28d24d24dd8ae8ac604217dff5d2b45cd3140957

**Review by pamodaaw — APPROVED** [review]

**Pull request merged**